src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -24,8 +25,23 @@ internal class ProcessMetrics
     internal static readonly AssemblyName AssemblyName = typeof(ProcessMetrics).Assembly.GetName();
     internal static readonly Meter MeterInstance = new(AssemblyName.Name, AssemblyName.Version.ToString());
 
+    private const string MetricPrefix = "process.dotnet.";
+
     static ProcessMetrics()
     {
+        // TODO: change to ObservableUpDownCounter
+        MeterInstance.CreateObservableGauge(
+            $"{MetricPrefix}memory.usage.physical",
+            () => Diagnostics.Process.GetCurrentProcess().WorkingSet64,
+            unit: "bytes",
+            description: "The amount of physical memory allocated for the current process.");
```

Comment:
Consider aligning with https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/process-metrics.md, and if there is something in the spec which should be improved, propose a change in the spec.